components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilder.java

```diff
@@ -418,17 +435,42 @@ public String getRelativeInternalURL() {
             return relativeInternalUrl;
         }
 
-        private String fetchAbsoluteUrl() throws URLBuilderException {
+        private String fetchAbsolutePublicUrl() throws URLBuilderException {
+
+            StringBuilder absolutePublicUrl = new StringBuilder();
+            if (StringUtils.isBlank(protocol)) {
+                throw new URLBuilderException("Protocol of service URL is not available");
+            }
+            if (StringUtils.isBlank(hostName)) {
+                throw new URLBuilderException("Hostname of service URL is not available");
+            }
+            absolutePublicUrl.append(protocol).append("://");
+            absolutePublicUrl.append(hostName.toLowerCase());
+            // If it's well known HTTPS port, skip adding port.
+            if (port != IdentityCoreConstants.DEFAULT_HTTPS_PORT) {
+                absolutePublicUrl.append(":").append(port);
+            }
+            absolutePublicUrl.append(fetchRelativePublicUrl());
+            return absolutePublicUrl.toString();
+        }
+
+        private String fetchAbsoluteInternalUrl() throws URLBuilderException {
 
-            StringBuilder absoluteUrl = new StringBuilder();
-            absoluteUrl.append(protocol).append("://");
-            absoluteUrl.append(hostName.toLowerCase());
+            StringBuilder absoluteInternalUrl = new StringBuilder();
+            if (StringUtils.isBlank(protocol)) {
+                throw new URLBuilderException("Protocol of service URL is not available");
+            }
+            if (StringUtils.isBlank(hostName)) {
```

Comment:
let's add the null check for internalHostName as well.

Reply:
Fixed https://github.com/wso2/carbon-identity-framework/pull/2949/commits/7a8092b90d4e91354b326e60bdef0c9b34ef10cb